Process_Library/GOCART2G_MieMod.F90

```diff
@@ -540,13 +541,15 @@ integer function getChannel(this, wavelength, rc) result (ch)
        endif
     enddo
 
-    if (present(rc)) rc = 0
-
-    if (ch < 0) then
-       !$omp critical (GetCha)
-       print*, "wavelength of ",wavelength, " is an invalid value."
-       !$omp end critical (GetCha)
-       if (present(rc)) rc = -1
+    if (present(rc)) then
+       if (ch > 0) then
+          rc = __SUCCESS__
+       else
+          rc = __FAIL__
+          !$omp critical (GetCha)
+          print*, "wavelength of ",wavelength, " is an invalid value."
+          !$omp end critical (GetCha)
+       endif
```

Comment:
```suggestion
    _ASSERT(ch > 0, "wavelength is invalid value.")
```

Reply:
A small bit of information is lost in the error message this way, so I defer  to @amdasilva as to whether this is an improvement.    Need to think about extending the error handling macros to allow more arguments.    CPP is just a bit too inflexible though, so these would need to become explicit calls and a bit more verbose.

Reply:
Never mind.   Regular MAPL macros are not available here.